tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerFactory.java

```diff
@@ -54,10 +51,21 @@ public static TinkerGraph createClassic() {
         return g;
     }
 
+    /**
+     * Create the "classic" graph with transaction support.
+     * @see #createClassic()
+     */
+    public static TinkerTransactionGraph createTxClassic() {
```

Comment:
It was done this way so that users could easily get either a transaction version or regular version of TinkerGraph without having to supply their own graph instance. This helps to keep it easy to use. Also, since there is currently only two implementations, I felt it was OK to just add Tx versions rather than completely making separate Factory classes for each of them. Though, I would agree that it makes sense to change how this works in the future if an implementation was added or removed.